Makefile

```diff
@@ -1,103 +1,187 @@
-# Expected Go version 1.12.x
-GO := go
-# Gofmt formats Go programs
-GOFMT := gofmt
-# Calculate cyclomatic complexities of Go functions
-# go get -u github.com/fzipp/gocyclo
-GOCYCLO := gocyclo
-# Golint is a linter for Go source code
-# go get -u golang.org/x/lint/golint
-GOLINT := golint
-# golangci-lint is a powerfull and fast meta-linter
-# https://github.com/golangci/golangci-lint
-GOLANGCI_LINT := golangci-lint
-GOLANGCI_LINT_CHECKERS := -D unused,staticcheck,errcheck,deadcode,structcheck,gosimple -E golint,gofmt
-
-# Commands to build by the build target.
-COMMANDS = $(shell ls cmd)
-
-# Protobuf compiler and code we might need to update/generate.
-PROTOC := protoc
-PROTOBUFS := $(shell find pkg -name \*.proto)
+# Go compiler/toolchain and extra related binaries we ues/need.
+GO_CMD    := go
+GO_FMT    := gofmt
+GO_CYCLO  := gocyclo
+GO_LINT   := golint
+GO_CILINT := golangci-lint
+
+# Disable some golangci_lint checkers for now until we have an more acceptable baseline...
+GO_CILINT_CHECKERS := -D unused,staticcheck,errcheck,deadcode,structcheck,gosimple -E golint,gofmt
+
+# Protoc compiler and protobuf definitions we might need to recompile.
+PROTOC    := protoc
+PROTOBUFS  = $(shell find cmd pkg -name \*.proto)
 PROTOCODE := $(patsubst %.proto,%.pb.go,$(PROTOBUFS))
 
-# Directories to docker build in, override tag to use, registry to docker push to.
-DOCKERDIRS = $(shell ls cmd/*/Dockerfile | sed 's:/Dockerfile::g')
-DOCKERTAG = ""
-DOCKERPUSH = ""
-# Path to kubelet binary
-KUBECTL=$(shell which kubectl)
-
-all: build # images
+# Binaries and directories for installation.
+BINDIR  := /usr/bin
```

Comment:
nit: should we have `PREFIX` as well? I.e. `BINDIR=$(PREFIX)/bin

Reply:
Thanks. Fixed.